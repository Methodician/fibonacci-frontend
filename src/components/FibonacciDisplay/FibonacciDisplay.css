.numberList {
    background-color: aqua;
    width: 90vw;
    display: flex;
    flex-wrap: wrap;
    align-self: center;
    margin-top: 20px;
}

.number {
    background-color: rgb(102, 56, 230);
    color: rgb(158, 244, 255);
    min-width: 20px;
    height: 25px;
    margin: 4px;
    text-align: center;
    padding: 0 3px 0 3px;
    font-weight: 500;
    border-radius: 10%;
}