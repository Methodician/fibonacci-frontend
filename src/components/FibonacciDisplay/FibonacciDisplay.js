import React from 'react';
import classes from './FibonacciDisplay.css';

const fibonacciDisplay = (props) => {
    if(props.numbers){
        return (
                <div className={classes.numberList}>
                    { props.numbers.map( (number, i) => {
                        return (
                            <div className={classes.number} key={i}>{number}</div>
                        );
                    })}
                </div>
                );
    } else return (<div></div>);
}

export default fibonacciDisplay;