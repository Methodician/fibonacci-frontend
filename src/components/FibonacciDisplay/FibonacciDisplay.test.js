import React from 'react';
import {shallow} from 'enzyme';
import FibonacciDisplay from './FibonacciDisplay';

test('should render fibonacci to 13 digits', () => {
    const display = shallow(<FibonacciDisplay numbers={[0,1,1,2,3]}></FibonacciDisplay>);

    console.log(display);

    expect(display.children.toString()).toEqual('hello');

});