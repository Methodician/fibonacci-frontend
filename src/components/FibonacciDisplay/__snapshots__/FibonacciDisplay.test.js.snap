// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render fibonacci to 13 digits 1`] = `
<div>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div>
    1
  </div>
  <div>
    2
  </div>
  <div>
    3
  </div>
</div>
`;
